src/PowerShellEditorServices/Services/DebugAdapter/DebugService.cs

```diff
@@ -603,6 +603,59 @@ public VariableScope[] GetVariableScopes(int stackFrameId)
             };
         }
 
+        internal void SetPathMappings(PathMapping[] pathMappings) => _pathMappings = pathMappings;
+
+        internal void UnsetPathMappings() => _pathMappings = null;
+
+        internal bool TryGetMappedLocalPath(string remotePath, out string localPath)
+        {
+            if (_pathMappings is not null)
+            {
+                foreach (PathMapping mapping in _pathMappings)
+                {
+                    if (string.IsNullOrWhiteSpace(mapping.LocalRoot) || string.IsNullOrWhiteSpace(mapping.RemoteRoot))
+                    {
+                        // If either path mapping is null, we can't map the path.
+                        continue;
+                    }
+
+                    if (remotePath.StartsWith(mapping.RemoteRoot, StringComparison.OrdinalIgnoreCase))
+                    {
+                        localPath = mapping.LocalRoot + remotePath.Substring(mapping.RemoteRoot.Length);
```

Comment:
It does seem to me like the mapping would also be a directory (which may or may not be supplied with a trailing path separator); is there a good reason _not_ to use path.combine?

Reply:
Root just means the root part of the path, this is how the python one works.If you prefer a different name I can certainly change it but:

+ I'm just working based on how the Python works and is named
+ I need to be able to specify just a name based on how Ansible works

Reply:
Ok, that's reasonable given Python's precedent.